### PR TITLE
Secondary PE Print for pe test 01

### LINKED ISSUE #303: Add few more prints to PE test 1 (Check Arch symmetry across PE)
We want to know the inconsistent value of PE registers for debugging. However, now the message only shows the masked out value. 

For verbosity message, add few prints to show values of registers right after the line https://github.com/ARM-software/bsa-acs/blob/main/test_pool/pe/operating_system/test_os_c001.c#L172.

For default mesasge, add prints right before the line https://github.com/ARM-software/bsa-acs/blob/main/test_pool/pe/operating_system/test_os_c001.c#L174.

### PR DESCRIPTION
**Pull request by ajayswar-s**
The PE test - 01 has been updated to check for all fails in registers and cache.
The secondary PE registers value can be stored in a structure and can be printed by Primary Pe for both fail and info cases.

Fixes #303, #304, #310

### CODE REVIEW

**Comment by sunnywang-arm** [comment]
Thanks for working on this, @ajayswar-s. 

Do you have the test result? For checking the messages from newly added prints, can we force the test to print the message for PASS case? 

I think this is for https://github.com/ARM-software/bsa-acs/issues/304

File: test_pool/pe/operating_system/test_os_c001.c [hunk 1]
```diff
@@ -243,28 +293,61 @@ payload(uint32_t num_pe)
...
+             }
+             else if (pe_buffer->pe_cache[cache_index] != 0)
+             {
+                   val_print(ACS_PRINT_ERR, "\n       PE Index: %d", i);
+                   val_print(ACS_PRINT_ERR, ", cache index: %d", cache_index);
+                   val_print(ACS_PRINT_ERR, ", size read: %llx     FAIL\n",
+                                                             pe_buffer->pe_cache[cache_index]);
+                   val_print(ACS_PRINT_ERR, "          PRIMARY PE VALUE : %llx \n",
+                                              cache_list[cache_index] & (~reg_list[0].reg_mask));
+                   val_print(ACS_PRINT_ERR, "          CURRENT PE VALUE : %llx \n",
```
Comment (sunnywang-arm) [inline test_pool/pe/operating_system/test_os_c001.c:315]: Can we change this message to "Masked Current PE value"?

File: test_pool/pe/operating_system/test_os_c001.c [hunk 2]
```diff
@@ -243,28 +293,61 @@ payload(uint32_t num_pe)
...
+                 val_print(ACS_PRINT_INFO, ", size read: %llx\n",
+                                                             pe_buffer->pe_cache[cache_index]);
+             }
+             else if (pe_buffer->pe_cache[cache_index] != 0)
+             {
+                   val_print(ACS_PRINT_ERR, "\n       PE Index: %d", i);
+                   val_print(ACS_PRINT_ERR, ", cache index: %d", cache_index);
+                   val_print(ACS_PRINT_ERR, ", size read: %llx     FAIL\n",
+                                                             pe_buffer->pe_cache[cache_index]);
+                   val_print(ACS_PRINT_ERR, "          PRIMARY PE VALUE : %llx \n",
```
Comment (sunnywang-arm) [inline test_pool/pe/operating_system/test_os_c001.c:313]: Can we change this message to "Masked Primary PE value"?

File: test_pool/pe/operating_system/test_os_c001.c [hunk 3]
```diff
@@ -243,28 +293,61 @@ payload(uint32_t num_pe)
...
+                 val_print(ACS_PRINT_INFO, "\n       PE Index: %d, ", i);
+                 val_print(ACS_PRINT_INFO, reg_list[reg_index].reg_desc, 0);
+                 val_print(ACS_PRINT_INFO, "  : %llx\n", pe_buffer->reg_data[reg_index]);
+             } else  {
+                 val_print(ACS_PRINT_ERR, "\n       PE Index: %d, ", i);
+                 val_print(ACS_PRINT_ERR, reg_list[reg_index].reg_desc, 0);
+                 val_print(ACS_PRINT_ERR, "   : %llx    FAIL\n", pe_buffer->reg_data[reg_index]);
+                 val_print(ACS_PRINT_ERR, "          PRIMARY PE VALUE : %llx \n",
+                                              rd_data_array[reg_index] & (~reg_list[i].reg_mask));
+                 val_print(ACS_PRINT_ERR, "          CURRENT PE VALUE : %llx \n",
```
Comment (sunnywang-arm) [inline test_pool/pe/operating_system/test_os_c001.c:332]: Can we change this message to "Masked Current PE value"?

**Review by sunnywang-arm — CHANGES_REQUESTED** [review]
I just reviewed the log offline and had a few comments inline and below.

Can we change all %llx to 0x%016llx?

Cc @chetan-rathore

**Comment by sunnywang-arm** [comment]
For https://github.com/ARM-software/bsa-acs/issues/310, the MIDR_EL1 register dump is only printed out with the max verbosity. Can we print the message below (all PE's MIDR_EL1 dump together) with default verbosity at the beginning of PE test 1? so that we don't need to run bsa-acs again with max verbosity to check the core type for the failure. 
```
PE Index: 1, MIDR_EL1  : xxxxxxxxx
PE Index: 2, MIDR_EL1  : xxxxxxxxx
PE Index: 3, MIDR_EL1  : xxxxxxxxx
PE Index: 4, MIDR_EL1  : xxxxxxxxx
PE Index: 5, MIDR_EL1  : xxxxxxxxx
PE Index: 6, MIDR_EL1  : xxxxxxxxx
PE Index: 7, MIDR_EL1  : xxxxxxxxx
.....
```

File: test_pool/pe/operating_system/test_os_c001.c (hunk 1 above)
Comment (chetan-rathore) [reply test_pool/pe/operating_system/test_os_c001.c]: done

File: test_pool/pe/operating_system/test_os_c001.c (hunk 2 above)
Comment (chetan-rathore) [reply test_pool/pe/operating_system/test_os_c001.c]: done

File: test_pool/pe/operating_system/test_os_c001.c (hunk 3 above)
Comment (chetan-rathore) [reply test_pool/pe/operating_system/test_os_c001.c]: done

File: test_pool/pe/operating_system/test_os_c001.c [hunk 4]
```diff
@@ -243,28 +293,61 @@ payload(uint32_t num_pe)
...
+          for (int reg_index = 1; reg_index < NUM_OF_REGISTERS; reg_index++) {
+             if (!(pe_buffer->reg_status[reg_index])) {
+                 val_print(ACS_PRINT_INFO, "\n       PE Index: %d, ", i);
+                 val_print(ACS_PRINT_INFO, reg_list[reg_index].reg_desc, 0);
+                 val_print(ACS_PRINT_INFO, "  : %llx\n", pe_buffer->reg_data[reg_index]);
+             } else  {
+                 val_print(ACS_PRINT_ERR, "\n       PE Index: %d, ", i);
+                 val_print(ACS_PRINT_ERR, reg_list[reg_index].reg_desc, 0);
+                 val_print(ACS_PRINT_ERR, "   : %llx    FAIL\n", pe_buffer->reg_data[reg_index]);
+                 val_print(ACS_PRINT_ERR, "          PRIMARY PE VALUE : %llx \n",
```
[the remark this answers is not in the document]
Comment (chetan-rathore) [reply test_pool/pe/operating_system/test_os_c001.c]: done

**Comment by chetan-rathore** [comment]
Thanks @sunnywang-arm for the comments.
Its been taken care.

**Review by chetan-rathore — APPROVED** [review]

**Pull request merged**